cpp/src/preprocessing/quantize/detail/scalar.cuh

```diff
@@ -87,10 +87,11 @@ std::tuple<T, T> quantile_min_max(raft::resources const& res,
   raft::random::RngState rng(seed);
   size_t n_elements  = dataset.extent(0) * dataset.extent(1);
   size_t subset_size = std::min(max_num_samples, n_elements);
-  auto subset        = raft::make_device_vector<T>(res, subset_size);
-  auto dataset_view  = raft::make_device_vector_view<const T>(dataset.data_handle(), n_elements);
-  raft::random::sample_without_replacement(
-    res, rng, dataset_view, std::nullopt, subset.view(), std::nullopt);
+
+  // select subsample element-wise
+  auto dataset_view =
+    raft::make_device_matrix_view<const T>(dataset.data_handle(), n_elements, 1UL);
+  auto subset = raft::matrix::sample_rows(res, rng, dataset_view, subset_size);
```

Comment:
I wonder whether we should check the available memory for before we try to allocate, or just make error reporting more informative, like below.
```suggestion
  auto subset = make_device_matrix<T, int64_t>(res, 0, 0);
  try {
    subset = raft::matrix::sample_rows(res, rng, dataset_view, subset_size);
   } catch (raft::logic_error& e) {
     // Informative error message, e.g. user can provide scaling factor to avoid extra memory needed for subsampling.
     throw;
   }
```